pkg/services/authn/authnimpl/sync/user_sync.go

```diff
@@ -136,7 +136,7 @@ func (s *UserSync) SyncLastSeenHook(ctx context.Context, identity *authn.Identit
 
 	namespace, id := identity.NamespacedID()
 
-	if namespace != authn.NamespaceUser && namespace != authn.NamespaceServiceAccount {
+	if id <= 0 || (namespace != authn.NamespaceUser && namespace != authn.NamespaceServiceAccount) {
```

Comment:
Would checking this before parsing the namespace / id be better (e.g. [here](https://github.com/grafana/grafana/pull/72776/files#diff-e2023442b1c843b4f67486826f9d3fdd78d9258352c64e1a7181df092fe3fc9dR132))? That would skip split / creating an int from string parts.

Reply:
maybe -- but does not help if we have `user:0` or `user:-1`, so I think this is OK.

It would help with the specific image renderer token though

Reply:
alo the `.NamespacedID()` calls has error handling that returns -1, and I think this call is better to ignore

Reply:
Seems fine for now but it would be easier to understand the code if it was a separate block with a comment.